### PR TITLE
Do not set ClusterCIDR for KubeProxy when using CNI networking and kubeProxy.clusterCIDR is not set

### PR DESCRIPTION
**Pull request by dezmodue**

### CODE REVIEW

**Comment by olemarkus** [comment]
Wouldn't this approach pretty much break any existing cluster that use `cni: {}`?

A better approach may be to make `clusterCIDR` a pointer and then test for nil vs empty string. Nil would mean current behavior, while you would want to set this to empty string.

**Comment by dezmodue** [comment]
Thanks for your comment and sorry for the late reply.
I understand this would be a breaking change, the intention was to align the behaviour of kops for the specific use case of `cni: {}` where kops doesn't have a lot of information to make a good call on defaults and the user is expected to take full ownership of the chi installation.

I have now modified the PR as suggested, this changes the behaviour of `clusterCIDR: ""` from "get the defaults" to omit the `--cluster-cidr` argument for kube-proxy.

An alternative could be to have a specific value of `clusterCIDR` that has the same effect, something like `clusterCIDR: "omit"`, this wouldn't change in any way the current behaviour and just add an extra option.

File: pkg/model/components/kubeproxy.go [hunk 1]
```diff
@@ -68,8 +68,12 @@ func (b *KubeProxyOptionsBuilder) BuildOptions(o interface{}) error {
 		// If we're not using the AmazonVPC networking, and the KubeControllerMananger has
 		// a ClusterCIDR, use that because most networking plug ins draw pod IPs from this range.
 		if clusterSpec.Networking.AmazonVPC == nil && clusterSpec.KubeControllerManager != nil {
-			config.ClusterCIDR = clusterSpec.KubeControllerManager.ClusterCIDR
+			config.ClusterCIDR = &clusterSpec.KubeControllerManager.ClusterCIDR
 		}
+		// The user can explicitly omit --cluster-cidr from the kube-proxy configuration
+		// by setting spec.kubeProxy.clusterCIDR: ""
+	} else if *config.ClusterCIDR == "" {
+		config.ClusterCIDR = nil
```
Comment (johngmyers) [inline pkg/model/components/kubeproxy.go:76]: If something like this is necessary, it should be done just before turning into flags, in `KubeProxyBuilder`.

**Comment by johngmyers** [comment]
Please rebase and squash.

**Comment by johngmyers** [comment]
You need to run the `hack/update-expected.sh` script.

File: pkg/model/components/kubeproxy.go (hunk 1 above)
Comment (dezmodue) [reply pkg/model/components/kubeproxy.go]: Not necessary - removed

**Review by johngmyers — APPROVED** [review]

**Pull request merged**